core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1767,10 +1767,18 @@ private[spark] class DAGScheduler(
 
           // TODO: mark the executor as failed only if there were lots of fetch failures on it
           if (bmAddress != null) {
-            val hostToUnregisterOutputs = if (env.blockManager.externalShuffleServiceEnabled &&
-              unRegisterOutputOnHostOnFetchFailure) {
-              // We had a fetch failure with the external shuffle service, so we
-              // assume all shuffle data on the node is bad.
+            val externalShuffleServiceEnabled = env.blockManager.externalShuffleServiceEnabled
+            val isHostDecommissioned = taskScheduler
+              .getExecutorDecommissionInfo(bmAddress.executorId)
+              .exists(_.isHostDecommissioned)
+            // Host shuffle data is considered lost if:
+            // - If we know that the host was decommissioned
+            // - Or when `unRegisterOutputOnHostOnFetchFailure` is enabled and we had
+            //   a fetch failure with the external shuffle service, so we assume all
+            //   shuffle data on the node is bad.
+            val hostLost = isHostDecommissioned || (externalShuffleServiceEnabled &&
```

Comment:
I have two issues:

1) As the `unRegisterOutputOnHostOnFetchFailure` is a feature flag for the unregistering with the following description:

https://github.com/apache/spark/blob/a4ca355af8556e8c5948e492ef70ef0b48416dc4/core/src/main/scala/org/apache/spark/internal/config/package.scala#L794-L801

But with your change when the host is decommissioned the feature flag is not considered at all and the unregistering will happen. I see the benefit to do the unregistering but if you leave this condition as it is now then please update the feature flag description too.

2) Can we find a different name from `hostLost`? As the host can be lost even when  `unRegisterOutputOnHostOnFetchFailure` is false. Or just inline this val as the comment above must be sufficient.

Reply:
I think for #1 updating the description text is probably the best path forward, I don't think we want to use that config flag.

Reply:
I think the key issue is that the unregisterOutputOnHost is poorly named but I think that should be fixed in a separate PR.

A better name/description for this config should be: "assume that all executor outputs on host are lost if any executor triggers fetch failure with an external shuffle service". Basically its just a "Hint" that is saying when to treat the executor loss as the host loss.

To which, I introduce another wrinkle: Shuffle output on the host is now considered unconditionally lost if we know that the host is decommissioned OR this above hint is enabled. Note that I am not introducing any separate config for the first part of this disjunction. 

I really don't think I am perverting the original meaning of unRegisterOutputOnHost. I am simply tacking an additional condition on top of it.

I have tried to rework the code with some comments to illustrate the logic. But please do help me understand what would be a better description/name here. Thanks.

Reply:
Edit: @attilapiros thanks for bringing me to rework this. I think I get the main intention of your suggestion now: unRegisterOutputOnHostOnFetchFailure now uniformly applies to both normal fetch failures and decommissioning. So the description does not need to be reworked. Please take a look at the new version.